hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FSDataInputStream.java

```diff
@@ -258,4 +258,14 @@ public int read(long position, ByteBuffer buf) throws IOException {
     throw new UnsupportedOperationException("Byte-buffer pread unsupported " +
         "by input stream");
   }
+
+  @Override
+  public void readFully(long position, ByteBuffer buf) throws IOException {
+    if (in instanceof ByteBufferPositionedReadable) {
+      ((ByteBufferPositionedReadable) in).readFully(position, buf);
+    } else {
+      throw new UnsupportedOperationException("Byte-buffer pread " +
+              "unsupported by input stream");
```

Comment:
Similarly, would be nice to log the class name of the input stream.

Reply:
Done